### DISCUSSION TITLE
Get the current state of a group, not just the individual members

### DISCUSSION THREAD

**Discussion by sv87411** [Koenkk/zigbee2mqtt#14294] [Devices]
Didn't know whether to categorise this as Devices or General, so apologies if this is more a general question.

I use the very useful groups feature in zigbee2mqtt with NodeRed & MQTT. I use homeassistant as well but want to use completely native MQTT control of my devices in node red.

I have just noticed that while I can send an MQTT '/get' request for the state to a group, only the individual group members return a message.

E.g.

`z2m/group1` contains 2 devices 
- ` z2m/light1`
-  `z2m/light2`

I send a message with a payload of `{"state":""}` to `z2m/group1/get`

I get returned state messages from `z2m/light1` and `z2m/light2` but there is no state update received about `z2m/group1`

You therefore have a way to request the state of a group, but you have to parse all of the responses from it's members to find out the group's state.

I would like to use the group entity to control a number of lights from a single switch on a NodeRed dashboard, but I'm finding it difficult to test the state of the group (so I can update the switch states) during a NodeRed restart.

If `/get` is supported for groups, why doesn't the group report it's own state in response? I'm watching the broker for responses from `z2m/group1` but I see nothing, all I see is the `/get` request.

Or am I missing something?

**Comment by sv87411** [comment]
Does anyone have any insight into this at all? I don't want to raise a feature request because it feels like if the group entity supports `/get` requests I would have thought it should provide a response and not just pass the `/get` on to the group members for response. 

The Groups documentation here [https://www.zigbee2mqtt.io/guide/usage/groups.html#groups](url) doesn't even mention that `/get` is available for groups and yet it can be issued. So I was wondering if maybe I'm missing something else that hasn't been fully documented?

Any help or insight much appreciated because I'd really like a group to behave just like a device when a `/get` is issued to it.

Happy to raise a feature request if it's missing functionality.

Thanks!

**Comment by MatAtBread** [comment]
I also have an issue with groups failing (in my case, failing to be created). They also fail in the app that came with the devices (TuYa Smart TRVs). Do you have any links to the ZigBee group spec?

**Reply by Maxcrouz73** [reply]
Hello,

Did someone manage to have state of group? It seems possible as the documentation mention it.
But i can't see any mqtt event on group state when i do actions on a devices of a group.
It will be a very useful feature

**Comment by Koenkk** [comment]
Zigbee groups don't have a state, so this is not possible.

**Reply by Maxcrouz73** [reply]
Hello, i was wondering yes following this doc:

https://www.zigbee2mqtt.io/guide/usage/groups.html

Anyway it should be very useful to have a state on a group as deconz do.

**Reply by Koenkk** [reply]
Zigbee2MQTT does publish a state for a group like explained in the docs, the state is based on the state of its members (as groups itself don't have states)

**Reply by Maxcrouz73** [reply]
Ok i see, group does not have a state but zigbee2mqtt publish a group state based on the state of the devices inside.
So technically you know if at least one of devices inside is on or if every devices are off.
